src/Symfony/Bridge/Doctrine/Types/AbstractUidType.php

```diff
@@ -92,4 +97,10 @@ public function requiresSQLCommentHint(AbstractPlatform $platform): bool
     {
         return true;
     }
+
+    private function hasNativeGuidType(AbstractPlatform $platform): bool
```

Comment:
Another option is to detect whether the column definition fallbacks to a `CHAR(36)`, which would then support other platforms added later (or third-party):

```
$platform->getGuidTypeDeclarationSQL([]) !== $platform->getVarcharTypeDeclarationSQL(['fixed' => true, 'length' => 36]);
```

Reply:
Fun fact: `getVarcharTypeDeclarationSQL()` is deprecated as well. 😇

Reply:
Well, use `getStringTypeDeclarationSQL` when the method exists as that's the replacement API :smile: